apps/openchallenges/infra/cdk/openchallenges_infra_cdk/prod/app.py

```diff
@@ -19,11 +25,77 @@ def main() -> None:
     developer_name = get_developer_name()  # Will be None for prod
     stack_prefix = get_stack_prefix()
 
+    # Get optional configuration
+    vpc_cidr = os.getenv("VPC_CIDR", "10.0.0.0/16")
+    certificate_arn = os.getenv("CERTIFICATE_ARN", "")
+    max_azs = int(os.getenv("MAX_AZS", "2"))  # Number of Availability Zones
+    app_version = os.getenv("APP_VERSION", "latest")
+    # FQDN is optional - if not provided, will use ALB DNS name
+    fqdn = os.getenv("FQDN", "")
+
     # Add common tags
     cdk.Tags.of(app).add("Environment", environment)
     cdk.Tags.of(app).add("Product", "OpenChallenges")
     cdk.Tags.of(app).add("ManagedBy", "CDK")
 
+    # Create VPC stack
+    # Production uses NAT gateways matching the number of AZs for high availability
+    vpc_stack = VpcStack(
+        app,
+        f"{stack_prefix}-vpc",
+        stack_prefix=stack_prefix,
+        environment=environment,
+        vpc_cidr=vpc_cidr,
+        max_azs=max_azs,
+        nat_gateways=max_azs,  # One NAT per AZ for high availability
```

Comment:
you can probably refactor these parameters to be optional on the condition that `MAX_AZS` is set.  then you can use the same code for every environment?

Reply:
Good suggestions! For now I want to develop the dev app first, then update the stage and prod app. At that time, I will refactor the code to promote DRY.